ApiClient/Source/Pages/Queries/SearchQuery.swift

```diff
@@ -27,22 +27,26 @@ import Foundation
 import Alamofire
 
 final class SearchQuery: Query {
-    typealias Result = JSONPage?
+    typealias Result = JSONPage
     
-    let resource: String
+    let url: URL?
+    let host: String = "https://api.schedjoules.com/"
     let method: HTTPMethod = .get
     let parameters: Parameters = [:]
     let headers: HTTPHeaders = ["Accept" : "application/json", "Content-Type" : "application/json"]
     
-    required init(resource: String) {
-        self.resource = resource
+    required init(path: String, queryItems: [URLQueryItem]) {
+        var urlComponents = URLComponents(string: host)
```

Comment:
is it possible to write this like so?
```swift
self.url = URLComponents(string: host, path: path, queryItems: queryItems).url
```

Reply:
No, unfortunately there is `URLComponents` does not have an initializer like that. Please see:
https://developer.apple.com/documentation/foundation/urlcomponents

Reply:
That's too bad. Apparently Apple doesn't understand the concept of immutability.